cai_causal_graph/causal_graph.py

```diff
@@ -614,6 +633,8 @@ def get_identifier(self) -> str:
 
     def is_dag(self) -> bool:
         """Check whether the `cai_causal_graph.causal_graph.CausalGraph` instance is a Directed Acyclic Graph (DAG)."""
+        if self._is_dag is not None:
+            return self._is_dag
         return networkx.is_directed_acyclic_graph(self.to_networkx()) if self._is_fully_directed() else False
```

Comment:
Should we cache `self._is_fully_directed()`, `self._is_fully_undirected()`, and `self.to_networkx()` as well? Could actually cache other `to_` methods but think it is overkill. The networkx one may be useful though.

Reply:
Done - cached `to_networkx` and `adjacency_matrix` too.